src/decays/decay_functions.cpp

```diff
@@ -0,0 +1,337 @@
+         result *= alpha_s_red;
+      case 0:
+         break;
+      default:
+         WARNING("Unknow correction in calc_Deltaqq");
+         break;
+   }
+
+   // order alpha_s_red^1 is taken into account with mass dependence somewhere else
+   return result;
+}
+
+/// Eq.(2.31) of hep-ph/0503172, including edge cases
+double RT(double x) noexcept
+{
+   if (x < 0.25) {
+      return RT_general(x);
+   } else if (x == 0.25) {
+      return std::numeric_limits<double>::quiet_NaN();
+   } else if (x < 0.95) {
+      return RT_general(x);
+   } else if (x < 1) {
+      const double d = x - 1;
+      const double d2 = d*d;
+      const double d4 = d2*d2;
+      const double d5 = d4*d;
+      return d5*(-3./10. + d*(13./20. + d*(-15./14. + d*(447./280. + d*(-95./42. + 523.*d/168)))));
+   } else if (x == 1) {
+      return 0;
+   } else if (x < 1.01) {
+      const double d = x - 1;
+      return d*(39 + d*(75.0/2.0 + d*(-6 + 5.0/4.0*d)));
+   }
+   return RT_general(x);
+}
+
+std::complex<double> f(double tau) noexcept {
+   if (tau <= 1) {
+      return Sqr(std::asin(std::sqrt(tau)));
+   }
+   else {
+      return -0.25*Sqr(std::log((1.+std::sqrt(1.-1./tau))/(1.-std::sqrt(1.-1./tau))) - 1i*Pi);
+   }
+}
+
+std::complex<double> taum1fprime(double tau) noexcept {
+   if (is_zero(tau-1)) {
+      return 0.;
+   }
+   else if (tau < 1) {
+      return (tau-1.)*std::asin(std::sqrt(tau))/std::sqrt(tau - Sqr(tau));
+   }
+   else {
+      return
+(tau-1.)*(1i*std::log(-1. - 1i*Pi + 2.*tau + 2.*std::sqrt((-1. + tau)*tau)))/
+   (-2i*std::sqrt((-1. + tau)*tau) + 
+     2*Pi*(2*(-1 + tau)*tau + std::sqrt((-1 + tau)*tau) - 2*std::sqrt((-1 + tau)*Power3(tau))));
+   }
+}
+
+std::complex<double> fprime(double tau) noexcept {
+   if (tau < 1) {
+      return std::asin(std::sqrt(tau))/std::sqrt(tau - Sqr(tau));
+   }
+   else {
+      return
+(1i*std::log(-1. - 1i*Pi + 2.*tau + 2.*std::sqrt((-1. + tau)*tau)))/
+   (-2i*std::sqrt((-1. + tau)*tau) + 
+     2*Pi*(2*(-1 + tau)*tau + std::sqrt((-1 + tau)*tau) - 2*std::sqrt((-1 + tau)*Power3(tau))));
+   }
+}
+
+// eq. 2.5, 2.7 & 2.8 of https://arxiv.org/pdf/hep-ph/0509189.pdf
+std::complex<double> delta_hAA_2loopQCD_for_quark_loop(double mH, double mq, double mu) noexcept
+{
+   const double tau = Sqr(mH/(2.*mq));
+   const std::complex<double> z = is_zero(tau) ? 1. : (std::sqrt(std::complex<double>(1. - 1./tau))-1.)/(std::sqrt(std::complex<double>(1.-1./tau))+1.);
+   const std::complex<double> ln = std::log(z);
+   const std::complex<double> li2p = dilog(z);
+   const std::complex<double> li2m = dilog(-z);
+   const std::complex<double> li3p = trilog(z);
+   const std::complex<double> li3m = trilog(-z);
+   const std::complex<double> li4p = Li4(z);
+   const std::complex<double> li4m = Li4(-z);
+
+   const std::complex<double> F0H = 3./(2.*Sqr(tau))*(tau+(tau-1.)*f(tau));
+
+   const std::complex<double> F0HC1H =
+      -z*(1. + z + Sqr(z) + Cube(z))/Power5(1.-z) * (108.*li4p + 144.*li4m - 64.*li3p*ln
```

Comment:
I haven't done any benchmarks, but there are a lot of redundant expressions in this code that could be calculated only once: `Power4(1.-z)` for example. But that can also be done later.

Reply:
Aren't those kinds of optimizations done automatically by the compiler? Especially since we've introduced `PowerN` as `constexpr` so they could be expanded.

Reply:
To some extend yes. The compiler will probably inline these `Power` functions. However, to my experience this is still slower than calculating these powers *once* and reuse them multiple times in the expression.

Reply:
Would you like me to replace every case of use of `PowerN(x)` for the same `N` and `X` that happens more than one or only the most common ones?

Reply:
Maybe you can at least replace `Power4(1.-z)` in this longish expression, where it is used many times. Going through all other expression is probably too excessive for now.

Reply:
Done in b595383d2b42b5c8e56f90a46be68b9e16aab37b

Reply:
Thanks a lot!